src/components/datepicker/common/datepicker-button.ts

```diff
@@ -0,0 +1,201 @@
+    if (formField) {
+      this.negative = isValidAttribute(formField, 'negative');
+
+      // We can't use getInputElement of SbbFormFieldElement as async awaiting is not supported in connectedCallback.
+      // We here only have to look for input.
+      const inputElement = formField.querySelector('input');
+
+      if (inputElement) {
+        this._inputDisabled =
+          isValidAttribute(inputElement, 'disabled') || isValidAttribute(inputElement, 'readonly');
+      }
+    }
+  }
+
+  private _init(picker?: string | SbbDatepickerElement): void {
+    this._datePickerController?.abort();
+    this._datePickerController = new AbortController();
+    this.datePickerElement = getDatePicker(this, picker);
+    this.setDisabledState(this.datePickerElement);
+    if (!this.datePickerElement) {
+      // If the component is attached to the DOM before the datepicker, it has to listen for the datepicker init,
+      // assuming that the two components share the same parent element.
+      this.parentElement?.addEventListener(
+        'inputUpdated',
+        (e: CustomEvent<InputUpdateEvent>) => this._init(e.target as SbbDatepickerElement),
+        { once: true, signal: this._datePickerController.signal },
+      );
+      return;
+    }
+    this._setAriaLabel();
+
+    this.datePickerElement.addEventListener(
+      'change',
+      (event: Event) => {
+        this.setDisabledState(event.target as SbbDatepickerElement);
+        this._setAriaLabel();
+      },
+      { signal: this._datePickerController.signal },
+    );
+    this.datePickerElement.addEventListener(
+      'datePickerUpdated',
+      (event: Event) => {
+        this.setDisabledState(event.target as SbbDatepickerElement);
+        this._setAriaLabel();
+      },
+      { signal: this._datePickerController.signal },
+    );
+    this.datePickerElement.addEventListener(
+      'inputUpdated',
+      (event: CustomEvent<InputUpdateEvent>) => {
+        this._inputDisabled = !!(event.detail.disabled || event.detail.readonly);
+        this._setAriaLabel();
+        this.onInputUpdated(event);
+      },
+      { signal: this._datePickerController.signal },
+    );
+
+    this.datePickerElement.dispatchEvent(datepickerControlRegisteredEventFactory());
+  }
+
+  private _setAriaLabel(): void {
+    const currentDate = this.datePickerElement?.getValueAsDate?.();
+
+    if (!currentDate || !this._dateAdapter.isValid(currentDate)) {
+      this.setAttribute('aria-label', this.i18nOffBoundaryDay[this._language.current]);
+      return;
+    }
+
+    const currentDateString =
+      this.datePickerElement?.now().toDateString() === currentDate.toDateString()
+        ? i18nToday[this._language.current].toLowerCase()
+        : this._dateAdapter.getAccessibilityFormatDate(currentDate);
+
+    this.setAttribute(
+      'aria-label',
+      this.i18nSelectOffBoundaryDay(currentDateString)[this._language.current],
+    );
+  }
+
+  private _setDisabledRenderAttributes(): void {
```

Comment:
would the disabled action mixin help here?

Reply:
I think not, because the component would inherit an unwanted `disabled` prop

Reply:
OK I see :)